File: gen_scripts/perl/f00.loop.pl
#!/usr/bin/env perl -w

# collection loop on list returned by subshell
foreach $item (split /\s+/, `ls dirtest`) {
  if (-d "dirtest/$item") {
    print "$item is a directory\n"
  } else {
    print "$item is not a directory\n"
  }
}

File: gen_scripts/ruby/f00.loop.rb
#!/usr/bin/env ruby
# testbox: requires=posix
# testbox: title="subshell with for..in collection"
# collection loop from output of subshell
for item in `ls dirtest`.split.each do   # cycle through directory listing
   if File.directory? "dirtest/#{item}"  # test if path is directory
       puts "#{item} is a directory"
   else
       puts "#{item} is not a directory"
   end
end
